core/domain/exp_services.py

```diff
@@ -1793,3 +1797,188 @@ def save_multi_exploration_math_rich_text_info_model(
+            new_state_dict['interaction']['hints']):
+        property_name_to_new_value_list.append(
+            (
+                exp_domain.STATE_PROPERTY_INTERACTION_HINTS,
+                new_state_dict['interaction']['hints']))
+    if old_state_dict['interaction']['solution'] != (
+            new_state_dict['interaction']['solution']):
+        property_name_to_new_value_list.append(
+            (
+                exp_domain.STATE_PROPERTY_INTERACTION_SOLUTION,
+                new_state_dict['interaction']['solution']))
+    if old_state_dict['interaction']['answer_groups'] != (
+            new_state_dict['interaction']['answer_groups']):
+        property_name_to_new_value_list.append(
+            (
+                exp_domain.STATE_PROPERTY_INTERACTION_ANSWER_GROUPS,
+                new_state_dict['interaction']['answer_groups']))
+
+    for property_name, new_value in property_name_to_new_value_list:
+        change_list.append(
+            exp_domain.ExplorationChange({
+                'cmd': exp_domain.CMD_EDIT_STATE_PROPERTY,
+                'state_name': state_name,
+                'property_name': property_name,
+                'new_value': new_value
+            }))
+    return change_list
+
+
+def get_batch_of_exps_for_latex_svg_generation():
+    """Returns a batch of LaTeX strings from explorations which have LaTeX
+    strings without SVGs.
+
+    TODO(#10045): Remove this function once all the math-rich text components in
+    explorations have a valid math SVG stored in the datastore.
+
+    Returns:
+        dict(str, list(str)). The dict having each key as an exp_id and value
+        as a list of LaTeX string. Each list has all the LaTeX strings in that
+        particular exploration ID.
+    """
+
+    latex_strings_mapping = {}
+    exploration_math_rich_text_info_models = (
+        exp_models.ExplorationMathRichTextInfoModel.get_all().filter(
+            exp_models. # pylint: disable=singleton-comparison
+            ExplorationMathRichTextInfoModel.
+            math_images_generation_required == True))
+    size_of_svgs_in_batch_bytes = 0
+    for model in exploration_math_rich_text_info_models:
+        if size_of_svgs_in_batch_bytes > (
+                feconf.MAX_SIZE_OF_MATH_SVGS_BATCH_BYTES):
+            break
+        latex_strings_mapping[model.id] = model.latex_strings_without_svg
+        size_of_svgs_in_batch_bytes += (
+            model.estimated_max_size_of_images_in_bytes)
+    return latex_strings_mapping
+
+
+def update_exploration_with_math_svgs(exp_id, image_data):
+    """Saves an SVG for each LaTeX string without an SVG in an exploration
+    and updates the exploration. Also the corresponding valid draft changes are
+    updated.
+
+    TODO(#10045): Remove this function once all the math-rich text components in
+    explorations have a valid math SVG stored in the datastore.
+
+    Args:
+        image_data: dict. The dictionary having all the image data like LaTeX
```

Comment:
Specify clearly the internal format of the dict. Give a full explanation of how the "dimensions" work.

Reply:
I've added a domain object for this